crates/wasmparser/src/readers/component/types.rs

```diff
@@ -226,6 +232,165 @@ impl PrimitiveValType {
         // component functions.
         a == b
     }
+
+    #[cfg(all(feature = "component-model", feature = "validate"))]
```

Comment:
To avoid the #[cfg] could this move inside of `mod component_types`? (or somewhere around there)